.github/workflows/CD.yml

```diff
@@ -1,45 +1,22 @@
-name: ci-cd
-
-on: [push, pull_request]
+name: CD
+
+on:
+  workflow_run:
+    workflows: [CI]
+    types:
+      - completed
+  
+  release:
+  types:
+    - created
 
 jobs:
-  ci:
-    # Set up operating system
-    runs-on: ubuntu-latest
-
-    # Define job steps
-    steps:
-    - name: Set up Python 3.8
-      uses: actions/setup-python@v4
-      with:
-        python-version: 3.8
-
-    - name: Check-out repository
-      uses: actions/checkout@v3
-
-    - name: Install poetry
-      uses: snok/install-poetry@v1
-
-    - name: Install package
-      run: poetry install
-
-    - name: Test with pytest
-      run: poetry run pytest tests/ --cov=cardsort --cov-report=xml
-
-    - name: Use Codecov to track coverage
-      uses: codecov/codecov-action@v3
-      with:
-        files: ./coverage.xml # coverage report
-
-    - name: Build documentation
-      run: poetry run make html --directory docs/
-
   cd:
     # Only run this job if the "ci" job passes
     needs: ci
```

Comment:
Using "needs: ci" here is redundant since you are already enforcing it via "workflow_run" above

Reply:
Ah yes, that makes sense, thank you.